src/main/java/org/jabref/gui/entryeditor/fileannotationtab/FulltextSearchResultsTab.java

```diff
@@ -91,45 +94,49 @@ protected void bindToEntry(BibEntry entry) {
             documentViewerView = new DocumentViewerView();
         }
         this.entry = entry;
-        PdfSearchResults searchResults = searchQueryProperty.get().get().getRule().getFulltextResults(searchQueryProperty.get().get().getQuery(), entry);
-
         content.getChildren().clear();
 
-        if (searchResults.numSearchResults() == 0) {
-            content.getChildren().add(new Text(Localization.lang("No search matches.")));
-        }
-
-        // Iterate through files with search hits
-        for (Map.Entry<String, List<SearchResult>> resultsForPath : searchResults.getSearchResultsByPath().entrySet()) {
-            entry.getFiles().stream().filter(file -> file.getLink().equals(resultsForPath.getKey())).findFirst().ifPresent(linkedFile -> {
-                content.getChildren().addAll(createFileLink(linkedFile), lineSeparator());
-
-                // Iterate through pages (within file) with search hits
-                for (SearchResult searchResult : resultsForPath.getValue()) {
-                    for (String resultTextHtml : searchResult.getContentResultStringsHtml()) {
-                        content.getChildren().addAll(TooltipTextUtil.createTextsFromHtml(resultTextHtml.replace("</b> <b>", " ")));
-                        content.getChildren().addAll(new Text(System.lineSeparator()), lineSeparator(0.8), createPageLink(linkedFile, searchResult.getPageNumber()));
-                    }
-                    if (!searchResult.getAnnotationsResultStringsHtml().isEmpty()) {
-                        Text annotationsText = new Text(System.lineSeparator() + Localization.lang("Found matches in Annotations:") + System.lineSeparator() + System.lineSeparator());
-                        annotationsText.setStyle("-fx-font-style: italic;");
-                        content.getChildren().add(annotationsText);
-                    }
-                    for (String resultTextHtml : searchResult.getAnnotationsResultStringsHtml()) {
-                        content.getChildren().addAll(TooltipTextUtil.createTextsFromHtml(resultTextHtml.replace("</b> <b>", " ")));
-                        content.getChildren().addAll(new Text(System.lineSeparator()), lineSeparator(0.8), createPageLink(linkedFile, searchResult.getPageNumber()));
+        stateManager.activeSearchQuery(SearchType.NORMAL_SEARCH).get().ifPresent(searchQuery -> {
```

Comment:
Looks very weird to put that into a lambda expression. Also makes stack traces way longer. Maybe a simple if check is enough or better - fail fast strategy (if (!activeSearchQuery.isPresent()) { return; } )